docs/rfc/rfc-101-p2p-bad-peers-checktx.md

```diff
@@ -0,0 +1,436 @@
+In this case we need to define what repeatedly means. 
+
+For b) we need to understand what is the potential reason a transaction could never have been valid on one node, but passes `CheckTx` on another node.  
+We need to understand all the possible scenarios in which this can happen:
+
+1. What happens if a node is misconfigured and allows, for example, very large transactions into the mempool. 
+This node would then gossip these transactions and they would always fail on other nodes. 
+Is this a scenario where we want nodes to disconnect from this peer and ban it but do not consider it malicious?
+2. Are all other reasons for this to happen sign of malicious behaviour where a node explicitly lies? How can `CheckTx` pass on a valid node, but fail on another valid node with a `ResponseCheckTx.code > 1`?
+If such behaviour is only possible when a peer is malicious, should this peer be punished or banned forever? Note that 
+we cannot know whether a node is a validator in order for it to be punished. Gossiping this behaviour to other peers pro-actively
+also entails a different set of problems with it - how do we know we can trust peers who tell us to ban other peers. For these reasons, understanding the actual reason for these failures can be left for future work. 
+
+For now, we will disconnect and ban the peer regardless of the exact reason a transaction 
+is considered to never be valid. 
+
+#### **Banning for frequent CheckTx failures**
+
+If a node sends transactions that fail `CheckTx` but could be valid at some point, a peer should not be banned the first time this happens.
+Only if this happens frequently enough should this be considered as spam. To define this behaviour we keep track how many times (`numFailures`) a peer 
+sent us invalid transactions within a time interval (`lastFailure`). This time interval should be reset every `failureResetInterval`. 
+
+For each peer, we should have a separate `numFailures` and `lastFailure` variable. There is no need to  have one per transaction.
+ Whenever a transaction fails, if the `now - lastFailure <= failureResetInterval`, we increment the `numFailures` for this particular peer and set the `lastFailure` to `now`. 
+ Otherwise, we set `lastFailure` to `now` and set `numFailures`  to 1. 
+ Once the value for `numFailures` for a peer reaches `maxAllowedFailures`, the peer is disconnected from and banned. 
+
+The reason for this logic is as follows: We deem it acceptable if every now and then a peer sends us an invalid transaction.
+ But if this happens very frequently, then this behaviour can be considered as spamming and we want to disconnect from the peer. 
+  
+ **Discussion**
+
+ The problem with supporting this scenario is the definition of the above mentioned parameters. It is very hard to estimate, at the CometBFT level, what these parameters should be. A possible solution is 
```

Comment:
I agree, I'm generally concerned about how we want to reason about this. I think its probably useful to iterate towards this, but also expose configs so node operators can adjust.

Also this type of ban, should likely always be timebound